### ISSUE TITLE
Keyboard input not being captured by search fields.

### ISSUE THREAD

**Issue by Zerophase** [ValveSoftware/steam-for-linux#5738]
#### Your system information

* Steam client version (build number or date):  1536359114
* Distribution (e.g. Ubuntu):  Arch Linux
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Typing in search fields does not capture input. I can't search game's owned or the store. Issue exists in both native and steam runtime.

#### Steps for reproducing this issue:

1.  Install Arch Linux.
2.  Setup cinnamon. (not sure if the desktop matters)
3. Launch steam.

**Comment by RyuueikenLiu** [comment]
Me too.It seems that no plug-ins calls Fictx or IBus for inputting. Maybe we need a plug-ins to call Fictx or IBus.

**Comment by RyuueikenLiu** [comment]
By the way ,the issue happens on when you input Chinese,Japanese or Korean which language inputting based on IME. Sorry,my English is not very well.Do you understand what I said?

**Comment by nanawel** [comment]
Ha, finally found people with this issue too.
I don't know if it's the same for you, but for me keyboard does not work at all with Steam and Steam games. I cannot even type my password in the login box. And if I use copy-paste (with right-click of course), I cannot then use any input text box on the store, the community, etc.

Then in games I cannot enter the name of my savegames either. But some functions seem to work, like camera control in Cities Skylines (pan/tilt).

It was working perfectly until some date around mid-september of this year, after an upgrade (but I don't know which one).

I have an up-to-date Archlinux with Steam build 1536436120.

**Comment by anthraxx** [comment]
Works perfectly fine here, but i would recommend that people who write "me too" basically provide some more info about their environments to find possible similarity.
desktop-environment, window-manager, compositor, x11/wayland

**Comment by Zerophase** [comment]
Yeah, I don't think it's working in games for me either. Just text input does not work. All keybutton presses work.

**Comment by nanawel** [comment]
> Works perfectly fine here, but i would recommend that people who write "me too" basically provide some more info about their environments to find possible similarity.
> desktop-environment, window-manager, compositor, x11/wayland

Sure.
So I'm using **XFCE** (4.12) with **xfwm4** (4.12.5) on **X11** (1.20.2).

**Comment by RyuueikenLiu** [comment]
I'm using GNOME with Deepin WM on Cantarell 11.

**Comment by RyuueikenLiu** [comment]
> Yeah, I don't think it's working in games for me either. Just text input does not work. All keybutton presses work.

Does your inputting base on  IME such as Fictx or IBus? I need to imput Chinese in the boxes,but it does not work.If I input English,that's OK.So the issue I encountered is that the inputting does not call the IME.

**Comment by Zerophase** [comment]
>

> > Yeah, I don't think it's working in games for me either. Just text input does not work. All keybutton presses work.
> 
> Does your inputting base on IME such as Fictx or IBus? I need to imput Chinese in the boxes,but it does not work.If I input English,that's OK.So maybe the issue I encountered is that the inputting does not call the IME.

Nope, just using traditional latin character inputs.

**Comment by nanawel** [comment]
> > Yeah, I don't think it's working in games for me either. Just text input does not work. All keybutton presses work.
> 
> Does your inputting base on IME such as Fictx or IBus? I need to imput Chinese in the boxes,but it does not work.If I input English,that's OK.So maybe the issue I encountered is that the inputting does not call the IME.

Same for me, I'm just using fr-latin1.
BTW having ibus daemon enabled or disabled doesn't make any difference (just in case it could matter).

**Comment by nanawel** [comment]
I can also add that for example I'm able to use the keyboard for movements and actions in L4D2. But again here I cannot type text anywhere. I thought it was important to mention this.

**Comment by Zerophase** [comment]
Here's the steam log from [native](https://pastebin.com/dtXcwzcd) and [runtime](https://pastebin.com/vUNi6rWm).

**Comment by Zerophase** [comment]
Well, rebuilt my system in a vm copy of my system, and the issue with keyboard input never cropped up.

**Comment by nanawel** [comment]
That's what I wanted to do too, but I expected the same result as you got. There must be a misconfiguration somewhere, but I can't figure it out.
I'm really annoyed by this issue, since it doesn't seem like it's going to be fixed by any upgrade so far.

**Comment by nanawel** [comment]
I've created a fresh new user, using Gnome instead of Xfce and with a *lot* less of background tasks but I can also reproduce the problem with it. So I guess we can say it likely comes from the system, not from the user session.

**Comment by Zerophase** [comment]
If anyone has this problem and is using the proprietary nvidia drivers, maybe try switching to nuevo, and seeing if that fixes it.

**Comment by nanawel** [comment]
> If anyone has this problem and is using the proprietary nvidia drivers, maybe try switching to nuevo, and seeing if that fixes it.

Well, in the event it would fix the issue indeed, I wouldn't be able to play games anymore with this driver at it still lacks decent performances. So... meh.

But thanks for bumping this thread, I'm dying here with my unusable Steam ;)

**Comment by Zerophase** [comment]
> > If anyone has this problem and is using the proprietary nvidia drivers, maybe try switching to nuevo, and seeing if that fixes it.
> 
> Well, in the event it would fix the issue indeed, I wouldn't be able to play games anymore with this driver at it still lacks decent performances. So... meh.
> 
> But thanks for bumping this thread, I'm dying here with my unusable Steam ;)

I'm just trying to figure out if it is a driver issue, considering the nvidia driver is not loaded in a vm.

**Comment by nanawel** [comment]
Yes sure, no offense of course.

**Comment by RyuueikenLiu** [comment]
In my situation, maybe I just need a Chinese input method plugin for inputting Chinese.

**Comment by nanawel** [comment]
Bumping a bit this thread, I have another test to add: using the virtual keyboard ([OnBoard](https://launchpad.net/onboard) in my case) does **not** work either. So I cannot type letters on any Steam window even with a virtual keyboard!

**Comment by mshade** [comment]
I had this issue as well and for me, it seems to have been caused by missing locale information. I had added `NoExtract` settings in `pacman.conf` to try to skip installing unnecessary language support. Turns out some were necessary :)

After removing the `NoExtract` options and reinstalling all system packages, Steam input now works. Hooray!

To do the same, clean out your `pacman.conf` and then:

```
# pacman -Qqn | pacman -S -
```

It was actually `dmenu_run` that lead me to the root cause. It segfaulted when text was entered, leading me to the missing locale files, and the `NoExtract` option I had added months ago. It seems that the issue only pops up after certain packages are updated and locale files begin to go missing - tricky!

**Comment by nanawel** [comment]
Hooo that does seem right @mshade!
I did not notice at the time but the issue must have arisen when I added a bunch of items in the `NoExtract` section (locales, to save space). But as you say, it did not appear immediately, but probably after some upgrades several days later.

I'll check that asap. Thank you very much for your help.
It would be even better to know exactly what locale files are necessary so that we could still use some `NoExtract` instructions to save space.

**Comment by kisak-valve** [comment]
Hello @Zerophase, are you still experiencing this issue on an up to date system? If you are, please share the output of `locale -a`. In particular, I'd like to know if `en_US.utf8` or possibly `C.utf8` / `C` are in the list.

**Comment by nanawel** [comment]
At least for my case, @mshade's solution worked perfectly, and I had no other issue after reinstalling all packages without skipping locales.

**Comment by Zerophase** [comment]
@kisak-valve I had those packages on my system at the time. Steam works fine now that I installed all the locales. I believe there's a check somewhere in the code that requires foreign language packs or the search functionality does not work. I should have a back up of the old pacman.conf if that helps.

**Comment by kisak-valve** [comment]
We know that Steam wants for en_US.utf8 to be available, and but has a couple fallback options. It sounds like the system didn't provide any locale that Steam expected to be available and that's not a particularly common system config.

Hello @RyuueikenLiu, your issue is separate and being tracked at #781 and #3255.

**Comment by Zerophase** [comment]
It did. Every open source application, and other closed source app worked fine. It's a bug in Steam that requires multiple language packs installed. I don't know which specific language pack is not needed. There's probably a nesting issue somewhere. I can reproduce the issue in a couple days to help you track it down, and just doing a default install of Arch with my pacman conf will remove all language packs other than en_US.utf8, and C.utf8. (any other English related packs stay on the system as well)

I understand that not everyone does this. This would just improve the Linux ecosystem, which I believe benefits Valve long term.
